lib/finch.ex

```diff
@@ -1,8 +1,48 @@
 defmodule Finch do
   @moduledoc """
   Finch
+
+  Create your Finch HTTP Client like so:
+
+      defmodule MyFinch do
+        use Finch
```

Comment:
I would prefer if we did not tie Finch to modules. Instead, we could make it so `finch` always requires the name on start and APIs. Then we can start it like this in your sup tree:

```elixir
{Finch, name: MyFinch, pools: %{
   :default => %{size: 10, count: 1},
   {:https, "hex.pm", 443} => %{size: 20, count: 4}
}
```

And then `Finch.request(MyFinch, ...)`. If users want to wrap it though, it would be easy. Thoughts?

Reply:
I can definitely see the appear of not being tied to a module, but I went this way because I wanted to avoid calculating the supervisor and registry names on every single interaction. Maybe that cost is not significant though.

I think we would also have to pass the pool config in for every request too, so that the PoolManager has access to it when it is starting the pools.

Realizing now that this means I cannot merge the runtime and compile time config in Finch.start_link and expect it to be used when calling Finch.request.

Reply:
I guess passing in the pool config to every request wouldn't be that bad. And if a user really has a huge amount of {s, h, p} to configure, they could stick the pool config in a persistent_term, which we could mention in the docs

Reply:
> I think we would also have to pass the pool config in for every request too, so that the PoolManager has access to it when it is starting the pools.

An alternative would be to stash the configuration in an ets table or persistent term and look it up using the `:name`.

Reply:
ha, great minds!

Reply:
Ok, i'm going to remove the __using__ macro and stick the configuration in a persistent term in start_link

Reply:
> I can definitely see the appear of not being tied to a module, but I went this way because I wanted to avoid calculating the supervisor and registry names on every single interaction. 

This is definitely a good point. What I usually do in this case is to start a ETS table with the same name and store the values in there. Since you also have a Registry, note the registry supports metadata, so you can name the registry with the given `name` and store this information in the registry metadata. This way you don't need to manage a separate ets table nor the persistent term storage.